### ISSUE TITLE
Image stitching doesn't always work

### ISSUE THREAD

**Issue by ravilich86** [Insta360Develop/Insta360_OSC#34]
I'm shooting a tour, the parameters on the camera do not change during the shooting of the tour. Sometimes stitching does not occur and the image is saved as DualFisheye without stitching. I checked all the parameters, but did not find a pattern why this happens. Has anyone encountered something similar?

Below are all the requests

Camera: **Insta360 X3**
Firmware **1.0.83**

Request: `{"parameters":{"optionNames":["hdrSupport","photoStitchingSupport"]},"name":"camera.getOptions"}`
Response: `{"name":"camera.getOptions","state":"done","results":{"options":{"hdrSupport":["off","hdr"],"photoStitchingSupport":["none","ondevice"]}}}`

Request: `{"name":"camera.setOptions","parameters":{"options":{"hdr":"hdr","photoStitching":"ondevice","captureMode":"image"}}}`
Response: `{"name":"camera.setOptions","state":"done"}`

Request: `{"name":"camera.takePicture"}`
Response: `{"id":"010628","state":"inProgress","name":"camera.takePicture","progress":{"completion":0}}`

Request:  `{"id":"010628"}`
Response: `{"id":"010628","state":"inProgress","name":"camera.takePicture","progress":{"completion":0.5}}`

Request: `{"id":"010628"}`
Response: `{"id":"010628","state":"inProgress","name":"camera.takePicture","progress":{"completion":0.5}}`

Request: `{"id":"010628"}`
Response: `{"name":"camera.takePicture","state":"done","results":{"_localFileGroup":["/DCIM/Camera01/IMG_20231127_160425_00_069.jpg"],"fileUrl":"http://192.168.42.1:80/DCIM/Camera01/IMG_20231127_160425_00_069.jpg","_fileGroup":["http://192.168.42.1:80/DCIM/Camera01/IMG_20231127_160425_00_069.jpg"]}}`

**Comment by Cornelius000** [comment]
Yes i have the same problem, do you have a fix?

**Comment by ravilich86** [comment]
I forgot to indicate in the question above that the error occurred specifically when using the OSC protocol. We abandoned the use of the OSC protocol and switched to using the SDK. The documentation contains an example of stitching an HDR image. If you can’t figure it out, I can post the approximate code later

**Comment by Cornelius000** [comment]
Yeah my problem is with the osc API as well. I have already contacted the support if they can help me with this, but I am not too hopeful. Do you know how this sdk works? Does it also use the osc API?

**Comment by ravilich86** [comment]
You need to request the SDK through this [link](https://www.insta360.com/cn/sdk/record). Once approved, the SDK for all platforms will be available to you. The SDK is functional, though the documentation is quite sparse and contains very few examples.

[This](https://github.com/Insta360Develop/CameraSDK-iOS/issues/40) discussion covers how to integrate the SDK into XCode. Start by obtaining the SDK, and if you have any questions, I will try to answer them.

**Comment by ravilich86** [comment]
You need to request the SDK through this [link](https://www.insta360.com/cn/sdk/record). Once approved, the SDK for all platforms will be available to you. The SDK is functional, though the documentation is quite sparse and contains very few examples.

[This](https://github.com/Insta360Develop/CameraSDK-iOS/issues/40) discussion covers how to integrate the SDK into XCode. Start by obtaining the SDK, and if you have any questions, I will try to answer them.
